### ISSUE TITLE
Potential Type Error

### ISSUE THREAD

**Issue by pa-** [Art-of-WiFi/UniFi-API-client#247]
Hi,

we discovered a Bug which occured between version v1.1.90 ... v.1.1.98:

`create_voucher` has array as return type (https://github.com/Art-of-WiFi/UniFi-API-client/blob/master/src/Client.php#L2111)
but will return the result of `fetch_results` (https://github.com/Art-of-WiFi/UniFi-API-client/blob/master/src/Client.php#L3933) which will return an `array` or in some case will return `bool`. This will cause an type error in some cases.

I am not that much into the details, but maybe it would be better to throw an Exception which could be catched?
Something like:

```php
    public function create_voucher(
        int    $minutes,
        int    $count = 1,
        int    $quota = 0,
        string $note = '',
        int    $up = null,
        int    $down = null,
        int    $megabytes = null
    ): array
    {
        $payload = [
            'cmd'    => 'create-voucher',
            'expire' => $minutes,
            'n'      => $count,
            'quota'  => $quota,
        ];

        if (!empty($note)) {
            $payload['note'] = trim($note);
        }

        if (!is_null($up)) {
            $payload['up'] = $up;
        }

        if (!is_null($down)) {
            $payload['down'] = $down;
        }

        if (!is_null($megabytes)) {
            $payload['bytes'] = $megabytes;
        }

        $result =  $this->fetch_results('/api/s/' . $this->site . '/cmd/hotspot', $payload);
        if (!is_array($result)) {
            throw new Exception();
        }

        return $result;
    }
```

**Comment by malle-pietje** [comment]
Thanks for reporting. We have spent a lot of time to implement parameter types and return types throughout the class. We’ll fix this bug allow a false value in case of errors which is the default.

Regarding Exceptions; we initially (7-8 years ago) left out Exceptions to make it easier for devs to use the client class. With the 1.2 releases we plan to add Exceptions because this can be a breaking for existing code.

**Comment by malle-pietje** [comment]
Fixed in v1.1.100
